hashicorp/terraform-provider-tfe
# feat: proactively check for RUM before safe delete

## Description

This PR is adding a proactive check for resources under management during safe deletes. If there are any RUM detected it will preemptively fail without calling the safe delete API.

## Testing plan

1. Create a workspace without the force_delete.
2. Add resources to the workspace
3. Attempt to delete the workspace
4. Expect to see this error returned:
```Error: Error deleting workspace <workspace id>: This workspace has <number of resources> resources under management and must be force deleted by setting force_delete = true```

## External links

_Include any links here that might be helpful for people reviewing your PR. If there are none, feel free to delete this section._

- [Related PR](https://github.com/hashicorp/terraform-provider-tfe/pull/675)

## Output from acceptance tests

_Please run applicable acceptance tests locally and include the output here. See TESTS.md to learn how to run acceptance tests._

_If you are an external contributor, your contribution(s) will first be reviewed before running them against the project's CI pipeline._

```
=== RUN   TestTFEWorkspace_delete_withoutCanForceDeletePermission
2022/11/20 20:47:58 [DEBUG] Delete workspace ws-testing
2022/11/20 20:47:58 [DEBUG] Delete workspace ws-testing
2022/11/20 20:47:58 [DEBUG] Delete workspace ws-testing
--- PASS: TestTFEWorkspace_delete_withoutCanForceDeletePermission (0.69s)
PASS

Process finished with the exit code 0


...
```
